tests/suite/resources_utils.py

```diff
@@ -946,6 +946,19 @@ def create_ingress_controller(v1: CoreV1Api, apps_v1_api: AppsV1Api, cli_argumen
     else:
         name = create_daemon_set(apps_v1_api, namespace, dep)
     wait_until_all_pods_are_ready(v1, namespace)
+    
+    if prom_port:
+        port = V1ContainerPort(9113, None, None, "prometheus", "TCP")
+        print("------------------------- Enable 9113 port in IC -----------------------------------")
+        body = apps_v1_api.read_namespaced_deployment(name, namespace)
+        body.spec.template.spec.containers[0].ports.append(port)
+
+        if cli_arguments["deployment-type"] == "deployment":
+            apps_v1_api.patch_namespaced_deployment(name, namespace, body)
```

Comment:
I think this patching will result into k8s bring down the IC running pod and provisioning a new one. This seems unnecessary.  Can we just add the port `9113` to the original manifest?   

Also, it's not mandatory to define ports in the pod spec. Just having them in the service resource should work